docs/src/proposals/fee_transaction_priority.md

```diff
@@ -0,0 +1,210 @@
+---
+title: Fee Transaction Priority
+---
+
+Additional fees were introduced to transactions as a method to allow users to bid for priority for
+their transactions in the leader's queue.
+
+The fee priority of a transaction `T` can then be defined as `F(T)`, where `F(T)` is the "fee-per
+compute-unit", calculated by:
+
+`(additional_fee + base_fee) / requested_compute_units`
+
+To ensure users get fair priority based on their fee, the proposed scheduler for the leader must
+guarantee that given `T1` and `T2` in the pending queue, and `F(T1) > F(T2)`:
+
+1. `T1` should be considered for processing before `T2`
+2. If `T1` cannot be processed before `T2` because there's already a transaction currently being
+processed that contends on an account `A`, then `T2` should not be scheduled if it would grab
+any account locks needed by `T1`. This prevents lower fee transactions like `T2` from starving
+higher paying transactions like `T1`.
+
+
+### Transaction Pipeline
+
+Pipeline:
+1. Sigverify
+2. Scheduler
+3. BankingStage threads
+
+Transactions from sigverify are connected via a channel to the scheduler. The scheduler maintains
+`N` bi-directional channels with the `N` BankingStage threads, implemented as a pair of two
+unidirectional channels.
+
+The scheduler's job is to run an algorithm in which it determines how to schedule transactions
+received from sigverify to the `N` BankingStage threads. A transaction is scheduled to be executed
+on a particular BankingStage thread by sending that transaction to the thread via its associated
+channel.
+
+Once a BankingStage thread finishes processing a transaction `T` , it sends the `T` back
+to the scheduler via the same channel to signal of completion.
+
+### Scheduler Implementation
+
+The scheduler is the most complex piece of the above pipeline, its implementation is made up of a
+few pieces. Note for now, all these pieces are maintained by the single scheduler thread to avoid
+locking complexity.
+
+#### Components of the `Scheduler`:
+
+1. `default_transaction_queue` - A max-heap `BinaryHeap<Transaction>` that tracks all pending transactions.
+The priority in the heap is the additional fee of the transaction. Transactions are added to this queue
```

Comment:
The priority in the heap is the F(T) = fee-per-cu. Part of the calculation is `base_fee` that includes signature, write_lock, and compute_fee that all depends on current `bank`'s feature_set and fee_structure (in `bank::calculate_fee(...)` ). Probably need to pass leader's current bank to Scheduler somehow